lighthouse-core/gather/gatherers/console.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/**
+ * @fileoverview Gathers all entries logged to the console, including console API calls,
+ * exceptions, and browser reports (on violations, interventions, deprecations, etc.).
+ */
+
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+
+/**
+ * @param {LH.Crdp.Runtime.RemoteObject} obj
+ * @return {string}
+ */
+function remoteObjectToString(obj) {
+  if (obj.value) {
+    // Primitive values will simply store the object.
+    return obj.value;
```

Comment:
still need to coerce to string for numbers and booleans?

Reply:
Hmm good point. The type is actually `any` in https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#type-RemoteObject. Not sure why the compiler allowed it